src/Compilers/CSharp/Portable/Parser/AbstractLexer.cs

```diff
@@ -131,9 +131,18 @@ protected XmlSyntaxDiagnosticInfo MakeError(int position, int width, XmlParseErr
 
         private int GetLexemeOffsetFromPosition(int position)
         {
-            return position >= TextWindow.LexemeStartPosition ? position - TextWindow.LexemeStartPosition : position;
+            return position >= LexemeStartPosition ? position - LexemeStartPosition : position;
         }
 
+        protected string GetNonInternedLexemeText()
+            => TextWindow.GetText(intern: false);
+
+        protected string GetInternedLexemeText()
+            => TextWindow.GetText(intern: true);
```

Comment:
these helpers are here because GetText implicitly uses LexemeStartPosition.  Once that is removed from the text window itself, it will need to be passed in (as the start position to read from, up to the text window's current position).  So this means instead of having to update a huge number of sites, only this site is updated.